dev_guide/builds.adoc

```diff
@@ -519,9 +519,52 @@ customStrategy:
 ----
 ====
 
+[[build-inputs]]
+== Build Inputs
+
+There are several ways to provide content for the build to operate on:  
+
+* Git repositories
+* Inline Dockerfile definitions
+* Binary inputs
+* Content extracted from existing images
+
+These inputs can also be combined in a single build.  If multiple inputs are specified and include overlapping filenames, the inline Dockerfile takes precedence, thereby overwriting any file named Dockerfile provided by another input.  Next in precedence is content from existing images, and finally content from the git repository or binary input.  Binary input and Git repository are mutually exclusive inputs.
+
+When the build is run, a working directory is constructed and all input content is placed in the working directory (e.g. the input git repository is cloned into the working directory, files specified from input images are copied into the working directory using the target path).  Next the build process will `*cd*` into the `*contextDir*` if one is defined.  Next the inline `*Dockerfile*` (if) is written to the current directory.  Last, the content from the current directory will be provided to the build process for reference by the `*Dockerfile*`, `*assemble*` script, or custom builder logic.  This means any input content that resides outside the `*contextDir*` will be ignored by the build.
+
+Here is an example of a source definition that includes multiple input types and an explanation of how they are combined.  For more details
+on how each input type is defined, see the specific sections for each input type.
+
+====
+[source,yaml]
+----
+source:
+  git:
+    uri: https://github.com/openshift/ruby-hello-world.git <1>
+  images: 
+  - from: 
+      kind: ImageStreamTag
+      name: myinputimage:latest
+      namespace: mynamespace
+    paths: 
+    - destinationDir: app/dir/injected/dir <2>
+      sourcePath: /usr/lib/somefile.jar
+  contextDir: "app/dir" <3>
+  dockerfile: "FROM centos:7\nRUN yum install -y httpd" <4>
+
+----
+
+<1> The repository to be cloned into the working directory for the build
+<2> `*/usr/lib/somefile.jar*` from `*myinputimage*` will be stored in `*<workingdir>/app/dir/injected/dir*`
+<3> The working dir for the build will become `*<original_workingdir>/app/dir*`
+<4> A `*Dockerfile*` with this content will be created in `*<original_workingdir>/app/dir*`, overwriting any existing file with that name
+====
```

Comment:
@bparees This might be a little bit personal preference-ey, but I think the above example can be combined with the para above a little more. The para is a little hard to follow until the example gives it a little more context. The specific example filenames etc could be put into the para, which might make it a little easier to follow. This could mean the little para above about "Here is an example" could disappear as well. This would leave the footnotes of the example to be a little less specific. Let me know what you think.